src/scripts/setup_ezdeploy.sh

```diff
@@ -79,23 +79,50 @@ container_registry_path="$(realpath "${this_script_path}")/container-registry"
 "${this_script_path}/util/checkfordocker.sh"
 
 # check mandatory parameters
-for i in { $docker_strategy $mlz_config_subid $mlz_config_location $tf_environment $mlz_env_name $web_port }
-do
-  if [[ $i == "notset" ]]; then
-    error_log "ERROR: Missing required arguments. These arguments are mandatory: -d, -s, -l, -e, -z, -p"
-    usage
-    exit 1
-  fi
-done
+if [[ $docker_strategy != "export" ]]; then
+  for i in { $docker_strategy $mlz_config_subid $mlz_config_location $tf_environment $mlz_env_name $web_port }
+  do
+    if [[ $i == "notset" ]]; then
+      error_log "ERROR: Missing required arguments. These arguments are mandatory: -d, -s, -l, -e, -z, -p"
+      usage
+      exit 1
+    fi
+  done
+fi
 
 # check docker strategy
 if [[ $docker_strategy != "local" && \
       $docker_strategy != "build" && \
-      $docker_strategy != "load" ]]; then
+      $docker_strategy != "load" && \
+      $docker_strategy != "export" ]]; then
   error_log "ERROR: Unrecognized docker strategy detected. Must be 'local', 'build', or 'load'."
   exit 1
 fi
 
+# build/load, tag, and push image
+image_name="lzfront"
+image_tag="latest"
+
+if [[ $docker_strategy == "build" || $docker_strategy == "export" ]]; then
+  echo "INFO: building docker image"
+  docker build -t "${image_name}" "${src_path}"
+fi
+
+if [[ $docker_strategy == "export" ]]; then
+  echo "INFO: Saving docker image and compressing it before exiting."
+  docker save "${image_name}:${image_tag}" -o mlz.tar
+  zip mlz.zip mlz.tar
+  rm mlz.tar
+  echo "INFO: Compressed deployable archive is saved locally as mlz.zip."
+  exit
+fi
```

Comment:
What do you think about abstracting this into a separate script? It seems that 'setup_ezdeploy.sh' doesn't describe what is happening given this `docker_strategy`. We seem to be doing something like 'export_dockerimage.sh'. 

going from:
```plaintext
./setup_ezdeploy.sh --docker-strategy export
### move machines ###
./setup_ezdeploy.sh --docker-strategy load {args}
```

to this:
```plaintext
./export_dockerimage.sh
### move machines ###
./setup_ezdeploy.sh --docker-strategy load {args}
```

would make it explicit what work is done from the script name, remove the the need to make any decisions about validation on L82, and give us the flexibility in the future to pass in a different image source path or allow a user to specify an output file name.

Thoughts? I might misunderstand the use case.

Reply:
Good idea,  more streamlined this way,  implemented.